tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_typecast.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-FileCopyrightText: © 2023-24 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from functools import partial
+import tt_lib as ttl
+
+
+from tests.tt_eager.python_api_testing.sweep_tests import (
+    comparison_funcs,
+    generation_funcs,
+)
+from tests.tt_eager.python_api_testing.sweep_tests.run_pytorch_ci_tests import (
+    run_single_pytorch_test,
+)
+
+mem_configs = [
+    ttl.tensor.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.DRAM),
+    ttl.tensor.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.L1),
+]
+
+
+@pytest.mark.parametrize(
+    "output_type",
+    [
+        ttl.tensor.DataType.BFLOAT16,
+        ttl.tensor.DataType.BFLOAT8_B,
+        # ttl.tensor.DataType.FLOAT32,
+        # ttl.tensor.DataType.UINT16,
+        # ttl.tensor.DataType.UINT32,
+    ],
+)
+@pytest.mark.parametrize(
+    "input_type",
+    [
+        torch.float32,
+        torch.float16,
+        torch.bfloat16,
+    ],
+)
```

Comment:
These tests are not actually testing the typecast operator.

What you have specified as output dtype is just being used to create the tt input tensor with how typecast op in `tt_lib_ops.py` is set up. This means typecast is just doing a copy and no data conversion since no output dtype was specified.

I think you should keep/use pairs for input/output dtype for both pt and tt tensors and use/pass them appropriately.

Ex.
```
@pytest.mark.parametrize(
     "pt_output_dtype, tt_output_type",
     [
         [torch.bfloat16, ttl.tensor.DataType.BFLOAT16],
         [torch.float32, ttl.tensor.DataType.BFLOAT8_B], # whatever equivalent we pick for bfloat8_b
     ],
 )
@pytest.mark.parametrize(
     "pt_input_dtype, tt_input_type",
     [
         [torch.bfloat16, ttl.tensor.DataType.BFLOAT16],
         [torch.float32, ttl.tensor.DataType.BFLOAT8_B],
     ],
 )
```